lib/results.js

```diff
@@ -139,8 +144,8 @@ function encodeResult(res, count) {
     output += (res.ok ? 'ok ' : 'not ok ') + count;
     output += res.name ? ' ' + res.name.toString().replace(/\s+/g, ' ') : '';
 
-    if (res.skip) output += ' # SKIP';
-    else if (res.todo) output += ' # TODO';
+    if (res.skip) output += ' # SKIP' + (typeof res.skip === 'string' ? ' ' + res.skip : '');
+    else if (res.todo) output += ' # TODO' + (typeof res.todo === 'string' ? ' ' + res.todo : '');
```

Comment:
same here?

Reply:
This better suits with the TODO descriptions in https://testanything.org/tap-version-13-specification.html#todo-tests

It _maybe_ useful, these messages also get passed in the JSON results which can help in diagnostics on the consumer side.

Reply:
ah - so this adds the assertion message after the ‘’TODO” marker? some tests would help illustrate this :-)

Reply:
Will do